[CI] FullClusterRestartIT testSnapshotRestore {cluster=OLD} failing

**Build Scans:**
- [elasticsearch-periodic #4922 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/exk2imhylxdui)
- [elasticsearch-periodic #4917 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/o5fysbzdozt6g)
- [elasticsearch-periodic #4912 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/pku5dl5phyl3w)
- [elasticsearch-periodic #4907 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/xjoq4el4alnae)
- [elasticsearch-periodic #4902 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/fcmgmjtnjo42m)
- [elasticsearch-periodic #4897 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/sk5vp2wwgkfoy)
- [elasticsearch-periodic #4892 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/zhxk7kcyvflda)
- [elasticsearch-periodic #4887 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/5xb2bwz5hfgfi)
- [elasticsearch-periodic #4882 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/lvhkbqisb7r7e)
- [elasticsearch-periodic #4877 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/vpjwxnhw5aq3a)

**Reproduction Line:**
```
./gradlew ':qa:full-cluster-restart:v8.14.3#bwcTest' -Dtests.class="org.elasticsearch.upgrades.FullClusterRestartIT" -Dtests.method="testSnapshotRestore {cluster=OLD}" -Dtests.seed=2FF2FD2C916AC54F -Dtests.bwc=true -Dtests.locale=en-PR -Dtests.timezone=America/Belize -Druntime.java=22
```

**Applicable branches:**
8.15

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.upgrades.FullClusterRestartIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testSnapshotRestore%20%7Bcluster%3DOLD%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: (<[8.14.3]> or <[8505000]> or <[8.14.0-8.14.3]>)
     but: was <[8.14.0-8.14.2]>
```

**Issue Reasons:**
- [8.15] 19 consecutive failures in step 8.14.3_bwc
- [8.15] 19 failures in test testSnapshotRestore {cluster=OLD} (2.1% fail rate in 905 executions)
- [8.15] 19 failures in step 8.14.3_bwc (100.0% fail rate in 19 executions)
- [8.15] 19 failures in pipeline elasticsearch-periodic (95.0% fail rate in 20 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/FullClusterRestartIT.java
+++ b/qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/FullClusterRestartIT.java
@@ -81,6 +81,7 @@ import static org.elasticsearch.test.MapMatcher.matchesMap;
 import static org.elasticsearch.transport.RemoteClusterService.REMOTE_CLUSTER_COMPRESS;
 import static org.elasticsearch.xcontent.XContentFactory.jsonBuilder;
 import static org.hamcrest.Matchers.anyOf;
+import static org.hamcrest.Matchers.contains;
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
@@ -90,6 +91,7 @@ import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.notNullValue;
 import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.Matchers.startsWith;
 
 /**
  * Tests to run before and after a full cluster restart. This is run twice,
@@ -1286,12 +1288,16 @@ public class FullClusterRestartIT extends ParameterizedFullClusterRestartTestCas
         assertEquals(singletonList(snapshotName), XContentMapValues.extractValue("snapshots.snapshot", snapResponse));
         assertEquals(singletonList("SUCCESS"), XContentMapValues.extractValue("snapshots.state", snapResponse));
         // the format can change depending on the ES node version running & this test code running
+        // and if there's an in-progress release that hasn't been published yet,
+        // which could affect the top range of the index release version
+        String firstReleaseVersion = tookOnIndexVersion.toReleaseVersion().split("-")[0];
         assertThat(
-            XContentMapValues.extractValue("snapshots.version", snapResponse),
+            (Iterable<String>) XContentMapValues.extractValue("snapshots.version", snapResponse),
             anyOf(
-                equalTo(List.of(tookOnVersion)),
-                equalTo(List.of(tookOnIndexVersion.toString())),
-                equalTo(List.of(tookOnIndexVersion.toReleaseVersion()))
+                contains(tookOnVersion),
+                contains(tookOnIndexVersion.toString()),
+                contains(firstReleaseVersion),
+                contains(startsWith(firstReleaseVersion + "-"))
             )
         );
 

